agent/engine/service_connect/manager_linux.go

```diff
@@ -15,17 +15,84 @@ package serviceconnect
 
 import (
 	"fmt"
+	"os"
+	"path/filepath"
 
 	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
 	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
 	dockercontainer "github.com/docker/docker/api/types/container"
 )
 
+const (
+	defaultRelayPathContainer  = "/var/run/ecs/relay/"
+	defaultRelayPathHost       = "/var/run/ecs/instance/relay/"
+	defaultRelayFileName       = "envoy_xds.sock"
+	defaultRelayENV            = "APPMESH_XDS_ENDPOINT"
+	defaultStatusPathContainer = "/var/run/ecs/"
+	defaultStatusPathHostRoot  = "/var/run/ecs/"
+	defaultStatusFileName      = "appnet_admin.sock"
+	defaultStatusENV           = "APPNET_AGENT_UDS_PATH"
+)
+
 type manager struct {
+	relayPathContainer  string
+	relayPathHost       string
+	relayFileName       string
+	relayENV            string
+	statusPathContainer string
+	statusPathHostRoot  string
+	statusFileName      string
+	statusENV           string
 }
 
 func NewManager() Manager {
-	return &manager{}
+	return &manager{
+		relayPathContainer:  defaultRelayPathContainer,
+		relayPathHost:       defaultRelayPathHost,
+		relayFileName:       defaultRelayFileName,
+		relayENV:            defaultRelayENV,
+		statusPathContainer: defaultStatusPathContainer,
+		statusPathHostRoot:  defaultStatusPathHostRoot,
+		statusFileName:      defaultStatusFileName,
+		statusENV:           defaultStatusENV,
+	}
+}
+
+func IsServiceConnectEnabledTask(task *apitask.Task) bool {
+	return task.IsServiceConnectEnabled()
+}
+
+func (m *manager) initializeAgentContainer(task *apitask.Task, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
+	return m.initServiceConnectDirectoryMounts(task.GetID(), container, hostConfig)
+}
+
+func getBindMountMapping(hostDir, containerDir string) string {
+	return hostDir + ":" + containerDir
+}
+
+func (m *manager) initServiceConnectDirectoryMounts(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
+	instancePath := filepath.Join(m.statusPathHostRoot, taskId)
```

Comment:
nit - maybe rename `instancePath` to `statusPathHost` for consistency and readability

Reply:
sold